mlir/include/mlir/Dialect/MIGraphX/MIGraphXOps.td

```diff
@@ -282,7 +282,7 @@ def MIGraphX_FlattenOp :
 def MIGraphX_TransposeOp :
     MIGraphX_Op<"transpose">,
     Arguments<(ins AnyRankedTensor:$input,
-                   I64ArrayAttr:$dims
+                   ElementsAttr:$dims
```

Comment:
Why this charge? You can go for `array<i64: v1, v2, ...>`?

Reply:
lemme check

Reply:
With I64ArrayAttr it only seems to accept dims = [0:i64, 2:i64, 1:i64]

Reply:
However, that fails in TOSA conversion 

https://github.com/ROCmSoftwarePlatform/rocMLIR/blob/8b9684b977f6773a7f30f1bcfa2fb344c16fdfa6/mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.td#L41-L42

Reply:
with 

```
/home/mkarunar/work/llvm-project-mlir/mlir/test/fusion/e2e/mixr-gemm/mixr-gemm-transpose.e2e.mlir:11:10: error: 'arith.constant' op value must be an integer, float, or elements attribute
    %2 = "migraphx.transpose"(%0) {dims = [0:i64, 2:i64, 1:i64]} : (tensor<1x5x3xf32>)-> tensor<1x3x5xf32>
         ^
/home/mkarunar/work/llvm-project-mlir/mlir/test/fusion/e2e/mixr-gemm/mixr-gemm-transpose.e2e.mlir:11:10: note: see current operation: %1 = "arith.constant"() {value = [0, 2, 1]} : () -> none
```

Reply:
Ah, got it.

I figure our two options here are to either go for
- Using a NativeCodeCall in the rewrites to convert the array attribute to a tensor attribute (that's where I'm leaning); or
- Making that constraints a bit more specific then ElementsAttr - perhaps I64DenleElementsAttr or some such

Reply:
Cool! I ll have a quick look at 1) then...

Reply:
Done.